xbmc/peripherals/bus/virtual/PeripheralBusAddon.cpp

```diff
@@ -514,9 +462,9 @@ void CPeripheralBusAddon::PromptEnableAddons(const ADDON::VECADDONS& disabledAdd
   bool bAccepted = false;
 
   auto itAddon = std::find_if(disabledAddons.begin(), disabledAddons.end(),
-    [](const AddonPtr& addon)
+    [](const BinaryAddonBasePtr& addonInfo)
   {
-    return std::static_pointer_cast<CPeripheralAddon>(addon)->HasButtonMaps();
+    return addonInfo->Type(ADDON::ADDON_PERIPHERALDLL)->GetValue("@provides_buttonmaps").asBoolean();
```

Comment:
Not direct, it can be maybe a static function added on ./xbmc/peripherals/addons/PeripheralAddon.h who do this with pass of addonInfo as value.

This way was thought (on agile) to replace cpluff and allow a read of addon.xml value without creation of related addon class, in this case CPeripheralAddon. Also used here to allow read of values of several different instance types without creation of them.